crates/holochain/src/conductor/space.rs

```diff
@@ -241,6 +241,12 @@ impl Spaces {
             }
         };
 
+        // If node_agents_in_spaces is not yet initialized, we can't know anything about
+        // which cells are blocked, so avoid the race condition by returning false
+        if cell_ids.is_empty() {
+            return Ok(false);
+        }
```

Comment:
Okay merging this to fix the race condition. But, security-wise, is there any other case that might trigger this early exit that we actually *would* want the connection to be blocked? Such as all cells being stopped or uninstalled, or something?

Reply:
Well it seems the only way there can be no agents is if the p2p db has no agents that match the node id. This is probably happening because alice discovered bob before bob discovered alice. Bob doesn't have the info necessary to determine whether he has blocked alice, he won't know that until he discovers her.

@thedavidmeister @neonphog can you help think through this? What are the situations where bob can not know alice's agent info? Certainly when alice first joins, and in that case we should probably just let them talk. But if bob blocked alice previously, would he ever forget her agent info and run into this case?

Another option is to make this check return an Err, which will trigger a `MetaNetEvtAuth::UnauthorizedIgnore`. That might be more correct, but it would probably gunk up the works when a new agent joins if we don't fix the race condition, because then a new agent is likely to be left hanging by whoever they talk to, depending on who wins the race.

As this PR stands, it fixed the slow gossip issue I was seeing. But yes we need to make sure this isn't some kind of security loophole.

Reply:
@maackle - a bit of info you don't have i think is the "preflight" connection message which automatically exchanges all local agent infos for the space on connection establishment before the ban check is made.

There was a test that showed that this worked properly, but of course there could be corner cases not covered.

Reply:
I'll add a warning log about this when the agent list is empty, which will let me verify that the preflight is indeed not working, and will also help us follow up with it later.

Reply:
Yep, here's what I see:

If I start up conductor 1 first, commit some entries, and then spin up conductor 2 on the same network, as soon as conductor 2 joins, conductor 1's log contains 18 instances in quick succession of the warning I just put in about an empty agent list. After doing some more stuff, including stopping and starting nodes, I haven't seen this warning come up again. However I do see some scattered instances of "No agents to gossip with on the node". But gossip performance seems to be good, and no more closed connections.